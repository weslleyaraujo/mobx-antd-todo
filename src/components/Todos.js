// @flow
import React from 'react';
import { List, Tooltip } from 'antd';

import { type Todo } from '../todo-store';

function Todos({
  onDoubleClick,
  dataSource
}: {
  onDoubleClick: (...args: any) => void,
  dataSource: Array<Todo>
}) {
  return (
    <List
      bordered
      dataSource={dataSource}
      renderItem={todo => (
        <List.Item
          style={{
            cursor: 'pointer',
            userSelect: 'none',
            textDecoration: todo.status === 'done' ? 'line-through' : 'none'
          }}
          onDoubleClick={event => {
            event.preventDefault();
            onDoubleClick(todo);
          }}
        >
          <Tooltip title="Double click to change status">{todo.text}</Tooltip>
        </List.Item>
      )}
    />
  );
}

export default Todos;
