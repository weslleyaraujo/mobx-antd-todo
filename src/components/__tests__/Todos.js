import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import Todos from "../Todos";

it("renders Todos correctly", () => {
  const wrapper = shallow(<Todos />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
