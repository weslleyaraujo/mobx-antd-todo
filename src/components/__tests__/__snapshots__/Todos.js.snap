// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Todos correctly 1`] = `
<List
  bordered={true}
  dataSource={Array []}
  loading={false}
  pagination={false}
  prefixCls="ant-list"
  renderItem={[Function]}
  split={true}
/>
`;
